resources/2024-07-secretstore-feature-spec.md

```diff
@@ -0,0 +1,329 @@
+# Extend the use-cases of `Applications.Core/secretStores`
+
+* **Author**: Will Tsai (@willtsai)
+
+## Topic Summary
+<!-- A paragraph or two to summarize the topic area. Just define it in summary form so we all know what it is. -->
+
+This feature spec outlines the desired user experience for extending the use-cases of `Applications.Core/secretStores` to allow for referencing secret stores in `Applications.Core/containers`, `Applications.Core/extenders`, `Applications.Core/volumes`, `Applications.Datastores/*`, and `Applications.Messaging/*` resources. The goal is to enable developers to securely manage secrets for use in their applications by referencing `Applications.Core/secretStores` in their resources.
```

Comment:
Would we be able to reference them also in Dapr and other UDTs?

Reply:
good point. added some dapr resource specific features.